src/v/storage/kvstore.cc

```diff
@@ -470,6 +477,7 @@ ss::future<> kvstore::load_snapshot_from_reader(snapshot_reader& reader) {
           batch.header().header_crc));
     }
 
+    auto lock = co_await _db_mut.get_units();
```

Comment:
Since snapshot creation needed a yield, maybe this loop over the same number of records also does?